Marlin/src/module/planner.cpp

```diff
@@ -774,6 +779,21 @@ block_t* Planner::get_current_block() {
 
   return nullptr;
 }
+block_t* Planner::lookahead(uint8_t offset) {
+  // Get the number of moves in the planner queue so far
+  const uint8_t nr_moves = movesplanned();
+
+  // If there are any moves queued ...
+  if (nr_moves < offset) return nullptr;
```

Comment:
Is there an off by one error here? If there was only the current block in the queue then I think `nr_moves` would be 1. And if `offset` were 1 then I think the next block is being requested, but there isn't one. So `nullptr` should be returned.

Should the condition not be `nr_moves <= offset`?

Reply:
```cpp
static inline uint8_t Planner::movesplanned()
Number of moves currently in the planner including the busy block, if any
```
You are right! good eye